src/panels/config/integrations/ha-config-integrations.ts

```diff
@@ -496,12 +498,33 @@ class HaConfigIntegrations extends SubscribeMixin(LitElement) {
   }
 
   private async _fetchManifests() {
-    const manifests = {};
     const fetched = await fetchIntegrationManifests(this.hass);
+    // Make a copy so we can keep track of previously loaded manifests
+    // for discovered flows (which are not part of these results)
+    const manifests = { ...this._manifests };
     for (const manifest of fetched) manifests[manifest.domain] = manifest;
     this._manifests = manifests;
   }
 
+  private async _fetchManifest(domain: string) {
+    if (domain in this._manifests) {
+      return;
+    }
+    if (this._extraFetchedManifests) {
+      if (this._extraFetchedManifests.has(domain)) {
+        return;
+      }
+    } else {
+      this._extraFetchedManifests = new Set();
+    }
+    this._extraFetchedManifests.add(domain);
```

Comment:
Why do we keep track of this, to prevent double calls while we are already fetching?

Reply:
Yeah. If someone discovers 100 ESPHome devices, we should just stick with 1 call.